### ISSUE TITLE
Executive Council 2024 Election

### ISSUE THREAD

**Issue by fperez** [jupyter-governance/ec-team-compass#23]
In this issue we'll track progress and questions regarding the 2024 election of the Jupyter Executive Council (EC).  As of this writing (late December 2023) we're leaving this as a placeholder, and will return after the holiday pause with more details and information. The EC wanted to communicate that elections _are happening in January 2024_ so folks can consider their potential participation, nomination, etc.

A few reminders of our elections process:

* Any member of a Standing Committee, Working Group, or Subproject Council (the "Union of Councils" or "UoC") is eligible to serve in the Jupyter Executive Council (a call for nominations will come shortly).
* [Per our governance model](https://jupyter.org/governance/executive_council.html#council-membership-and-elections), we will have 3 open seats this election cycle. _Two will be chosen by the Executive Council and one will be chosen by the community_. 

Please feel free to ask any questions below, and stay tuned for a formal call for nominations and detailed timeline in the opening days of 2024. Happy end of 2023!

**Comment by fperez** [comment]
This issue was mentioned by @Ruv7 in today's [Discourse post](https://discourse.jupyter.org/t/jupyter-executive-council-election-2024/23042), we'll track further discussion here.

**Comment by ivanov** [comment]
> *Two will be chosen by the Executive Council and one will be chosen by the community.*

I'm curious about the reason for this. The [governance docs state](https://jupyter.org/governance/bootstrapping_executive_council.html) that

>  Once the full set of candidates is assembled, the UoC and then the EC will take ranked choice votes to fill their allocated number of EC seats. After each election, the total number of seats filled by the UoC should be equal to, or one more than, the number filled by the EC.

With Fernando, Brian, and Darian not counting toward the "towards the running total of members elected by the EC or Union of Councils" [per Bootstrapping Governance document](https://jupyter.org/governance/bootstrapping_executive_council.html), with three vacancies, it seems like there should be two chosen by the Union of Councils, and one the Executive Council, which would be a repeat of the way last year's election was run. 

It seems to me like next year (2025) should be the first time that there are two seats filled by the EC, and one by the UoC, because that will be the first time we can attribute all six members of the EC as being elected by the EC or the UoC.

Here's a visual that captures my understanding, with the bootstrapped trio as dashes (-), and indicating other members by the origin of their election (U for Union of Councils, E for Executive Council), laying out the composition of the Executive Council by the end of their term.

```
2023-2024: U U E     per last year's election, serving 1 year terms
2023-2025: - - -     bootstrapped trio, which does not count as either UoC or EC elected seats
2024-2026: U U E     we are here now, and again with 3 seats total, 2 should be UoC elected seats 
2025-2027: U E E     6 seats total, 3 should be UoC elected seats
```

**Comment by jasongrout** [comment]
Paul, thanks for posting here. I think I see the ambiguity you are pointing out. When the governance group originally wrote "total number of seats filled", we were trying to capture the idea that if you add up all the seats ever filled since the beginning of the EC (except the initial 3 members), you would have this balance between community and EC appointments, and that would lead to two E seats and one U seat in your diagram so that the totals across time are 3 Es and 3 Us.

But if I understand you correctly, you are reading the scope of "total" to be "total seats in a given EC at a specific point in time", which would lead to your conclusion that we should have 2 Us and one E this time. I don't know how often these two interpretations might have different outcomes, but you're right that for this election they are different.

**Comment by ivanov** [comment]
I wasn't in the room where it happened, but I disagree with the "total across time" interpretation. Last time, there were three seats, two of which where chosen by the Union of Councils, and all three of those seats are now being vacated. We still only have 3 seats that "count" in this upcoming election, and to meet the requirement  of

> After each election, the total number of seats filled by the UoC should be equal to, or one more than, the number filled by the EC.

We should not have the EC choosing more than the majority of seats, which is what would happen if we have the EC choose two out of the three seats this time.

**Comment by ivanov** [comment]
Another reason to have the Union of Councils select two of the three seats this time is because this selection is for a full two-year term, whereas this past year's two seats selected by the Union of Councils only got to serve a one year term.

**Comment by ivanov** [comment]
Jason and I again discussed this at [the Executive Council office hours today (February 22, 2024)](https://docs.google.com/document/d/1N5jm0VJlDUX4WK0n7Ia8Ff63RdzNYkH28PkSZPdBtUc/edit#heading=h.111gugd304uc). I am inlining those notes below 

 - Paul and Jason discussed[ Paul’s concerns about how many seats should the
   be allotted to the Executive Council versus the Union of Councils in the
   upcoming election](https://github.com/jupyter-governance/ec-team-compass/issues/23#issuecomment-1920059771)

   - Paul strongly believes that the Union of Councils should again select 2
     seats, and the Executive Council should again select 1 seat.

   - Jason understands the appeal of the “point in time” interpretation of
     the governance documentation, but during the drafting the model that was
     being proposed and discussed was “total across time”, he believes the
     minutes from those discussions reflect this. (see action items)

   - Both agree that this election would be the only time the two
     interpretation seem to differ in allocation

   - Hard to resolve ambiguity of what was the mental model people had when
     they approved the governance documents (what did they/we believe when we
     voted on it)

   - Jason proposes an advantage of “total across time” interpretation is
     that it never has to be revealed who a given EC member represents - were
     they elected by the UoC or the EC, in cases where a seat is vacated
     before the end of a term and needs to be refilled in the following
     election.

     - When joining EC, Jason stated that Darian provided a link to the
       newly-elected EC members to see the election results, but also
       encouraged them to not look in that they would be better off not
       knowing that and not feeling like they are filling a “community” or
       “executive council” appointed seat.

     - Paul understands the appeal of such an arrangement, but does not think
       this is a requirement, as elections have to be auditable and
       transparent to have the trust and legitimacy of all involved.

       - So while there are legitimate reasons for not wanting to broadcast
         the constituency of a given EC member, the election committee still
         knows, and at least the elected EC have the ability to know.

     - Paul proposes an alternative for dealing with not keeping track of or
       revealing the constituency of an EC member vacating their seat early,
       which is to consider that seat a blank seat - that is - neither filled
       by UoC nor the EC, same as the initial bootstrapped three seats.

   - Jason to go back to governance formation minutes and reply on the [issue 23](https://github.com/jupyter-governance/ec-team-compass/issues/23)

**Comment by ivanov** [comment]
It's been three weeks and no one has gotten back to me about this. I find this unwillingness to address the issue that I have raised mutliple times now rather frustrating, given that @afshin just sent out an email to the union of councils which still includes the wording 

>   Once three nominees are selected (one from the Union of Councils vote described above and two selected by the current EC)

which violates the wording of our governance document.

To repeat: The total number of seats filled by the union of councils will *not* "be equal to, or one more than, the number filled by the EC." Specifically, with three seats total, it will be one less than the number filled by the EC. With big changes like [the proposed move to Linux Foundation](https://github.com/jupyter/governance/issues/204), it is particularly important to allow the Union of Councils more than just one seat.

**Comment by ivanov** [comment]
I just sent an email to the Executive Council with content similar to the post above, CCing several subproject councils for awareness of this issue.

**Comment by jasongrout** [comment]
Paul, I apologize for not replying back here yet. As you may remember from our verbal discussion several weeks ago, the EC priority was getting the LF proposal out, and then I fell ill the day after that was published last week. Yesterday and today I'm gradually returning to work as I have energy. However, I should not have left you hanging from my action item to look back through the governance minutes. Sorry.

**tl;dr: The EC is running our current EC election in accordance with the current governance docs.**

The sentence in question from the [EC governance doc](https://jupyter.org/governance/executive_council.html#council-membership-and-elections) is:

> After each election, the total number of seats filled by the UoC should be equal to, or one more than, the number filled by the EC.

This discussion stems from interpreting the word "total" differently - @ivanov interprets it as meaning "current election total", while the EC interprets it as "running total" or "cumulative total".

As @ivanov summarized from our verbal discussion in EC office hours several weeks ago, 

> Jason understands the appeal of the “point in time” interpretation of the governance documentation, but during the drafting the model that was being proposed and discussed was “total across time”, he believes the minutes from those discussions reflect this. (see action items)

There has never been any doubt among those who wrote the sentence in question about the interpretation - the intent and meaning of "total" is "running total" or "cumulative total" rather than "point-in-time total", consistent with how the EC is currently running the election. In other words, after every election, the (cumulative) total of the number seats filled by the UoC (since the first EC election) should be equal to, or one more than, the number filled by the EC (since the first EC election). After the current EC election, we will have 2 (first election) + 1 (this election) = 3 seats filled by the UoC, and 1 (first election) + 2 (this election) = 3 seats filled by the EC.

This interpretation of "total" meaning "running total" is supported by the more complete phrase "running total" in the [EC Bootstrapping document](https://jupyter.org/governance/bootstrapping_executive_council.html#mechanics), approved as part of the current governance model:

> Fernando Pérez, Brian Granger, and Afshin Darian will start as initial members of the EC and will serve two-year terms. Their initial appointment does not count towards the running total of members elected by the EC or Union of Councils (UofC).

and [later in the same document](https://jupyter.org/governance/bootstrapping_executive_council.html#implications-questions), again we mention "running election tally":

> The choice of Fernando, Brian, and Darian as initial members is allocated by the current Steering Council upon voting for this document. This is why their initial membership does not count towards the running election tally of later members chosen either by the EC or the UofC.

Paul, I found these clarifying references when I was looking through the governance meeting minutes, but then I realized the language had carried through to the existing official docs as well. Hopefully that helps you see the intent of the language in the EC doc. Sorry I had forgotten about these references to "running total" and "running election tally" when we last talked.

That said, we can always change the existing governance docs if we want a different process. That would require a proposal and an approval vote of the EC and SSC.

**Comment by jasongrout** [comment]
Thanks @willingc and @ivanov for engaging in this discussion. I totally agree that we should definitely clarify the existing language, one way or the other. I'm also partial to Paul's interpretation as it weighs things toward the community (though I think it would be a change in process, so deserves a proposal and vote).

> If you do an annual governance bylaws review, these two points along with a third point: "who within the EC is able to appoint the next EC members when there are EC members on the ballot". Is it the EC members who remain who are not on the ballot? Or all of the EC members?

Thanks for the two excellent points - a regular review of the governance docs, as well as bringing up what I think would be a clear conflict of interest. I would hope that any EC members that are on the ballot for the next EC would recuse themselves from the selection process.

**Comment by mwcraig** [comment]
Sorry for the delay in joining the conversation -- despite a couple of years of experience reading a bargaining contract very closely during a time when I was union president and extensive layoffs were happening at my university, I found the language in the governance document confusing. As @willingc said, governance is hard.

I can see three readings of the language:

1. After each election, of the number of people serving on the EC, the number elected to the EC by the UoC must equal or exceed by one the number elected by the EC.
2. In a running total of all elections, the number of times the UoC has had a chance to vote must equal or exceed by one the number of times the EC has a chance to vote.
3. (I find this one implausible but mention it for completeness) In each election, the number of seats voted on by the UoC must equal or exceed by one the number of seats  voted on by the EC.

This sentence from "[Council membership and elections](https://jupyter.org/governance/executive_council.html#council-membership-and-elections)"  states that "After each election, the *total* [emphasis added] number of seats filled by the UoC should be equal to, or one more than, the number filled by the EC." This could mean either 1 or 2 above, and seems to me to exclude 3. The more natural reading to me is that it means 1.

The [bootstrapping process](https://jupyter.org/governance/bootstrapping_executive_council.html) seems to use clearer language, which goes out of its way to stress that the initial three EC members do not count towards the "running total", as Jason points out above, and contains the only specific allocation of seats that I saw (third bullet in the "Machanics" section): "In the first EC election, three additional people will be elected and will serve one-year terms. Two of these will be selected by the UofC election, and one will be elected by the initial EC."

This perhaps supports interpretation 2, though I think it is consistent with interpretation 1.



A few additional areas that I think could use clarification/suggestions for the future:

1. This sentence in ["Council membership and elections"](https://jupyter.org/governance/executive_council.html#council-membership-and-elections) "Once the full set of candidates is assembled, the UoC and then the EC will take ranked choice votes to fill their allocated number of EC seats." is the first place in the governance docs that allocation of seats is mentioned. It should be made clearer how those seats are allocated. The only other reference I found to allocation was in the bootstrapping section.
2. In the [Leadership Directory](https://jupyter.org/governance/people.html), for each person on the EC
     1.  list when their term expires
     2. state whether they were elected by the UoC or by the EC or neither

**Comment by Zsailer** [comment]
Hi all, I appreciate the dialogue here. I think there is more to review, discuss, and improve.

I'd like to close this issue, given that the original intent of the post was to track the election from last year. We should open a new thread to discuss improving language around vote tally, so that the discussion is easier to find/index. (I was surprised to see the depth of discussion when triaging issues. 😅 )

**Comment by jasongrout** [comment]
I've drafted a governance PR (https://github.com/jupyter/governance/pull/252) that tries to incorporate suggestions from this discussion and https://github.com/jupyter/governance/issues/251.

As Zach mentioned above, I'll close this issue, but please feel free to continue discussion on how to improve the governance docs at https://github.com/jupyter/governance/pull/252
